crates/tuono/src/route.rs

```diff
@@ -261,4 +262,12 @@ mod tests {
             assert_eq!(route.output_file_path(), PathBuf::from(html))
         }
     }
+
+    #[test]
+    fn should_correctly_replace_double_underscore() {
+        let dyn_info = AxumInfo::new(&Route::new("/[posts__id]".to_string()));
```

Comment:
I see a problem here. The user can define both `post__id` and `post_id` route, and they resolve with the same name.

How about replacing the `__` with the `u_` string?

Reply:
I think the best solution would be to remove the `_` from the replacement prefix. `catch_all_`and `hyphen_`

Reply:
hmm yeah that sounds more reasonable. let me remove the underscore prefix on the replacement logic

Reply:
one small issue is the generated name will be a bit harder to read for hypen
`"/posts/handle-this/[post]" => "posts_handlehyphen_this_dyn_post"`
the case I see where double underscore will happen on hypen if it's use at thestartd of route name in dynamic route or catch all route. where catch all does need dyn for it to work. i think it's fine to only change the `_catch_all_` to `catch_all_`

Reply:
It sounds the best. Agree